components/studio/bin/hab-studio.sh

```diff
@@ -1004,8 +1004,18 @@ persisted. Check that the filesystem is no longer in the mounted using \
 # if true and non-zero otherwise.
 is_fs_mounted() {
   _mount_point="${1:?}"
-
-  $bb mount | $bb grep -q "on $_mount_point type"
+  
+  # Work around bug in musl's implementation of getmntent_r. 
+  # https://github.com/habitat-sh/habitat/issues/6591#issuecomment-498292168 
+  #$bb mount | $bb grep -q "on $_mount_point type"
+
+  # NOTE(SM): There is still a chance for failure here. Mount points with a 
+  # space in their name will not be detected. However, given that we control
+  # the name of all mount points, it is unlikely that we will encounter this 
+  # scenario.  
+
+  # NOTE(SM): This makes this studio implementation Linux specific. 
+  $bb cut -d' ' -f2 /proc/mounts | $bb grep -q "$_mount_point" 
```

Comment:
I think I need to do a little more work here.  The grep previously matched `on $mount type` so was guaranteed to only match one item. We're splitting on whitespace from /proc/mount now so there is the potential for multiple matches.  If busybox grep supports it, I'm going to switch this to a literal match and verify it works.

cc @baumanj

Reply:
By way of example: 
```
[23][default:/src:130]# cut -d' ' -f2 /proc/mounts |grep -x /dev
/dev
[24][default:/src:0]# cut -d' ' -f2 /proc/mounts |grep /dev
/dev
/dev/pts
```

Reply:
That seems reasonable. The `-x` flag is part of [POSIX for grep](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/grep.html) and busybox does seem to support it